cmds/db-manager/main.go

```diff
@@ -0,0 +1,228 @@
+				log.Println("Stopping after this running migration ...")
+				myMigrater.GracefulStop <- true
+				return
+			}
+		}()
+	}
+	totalMove := 0
+	migrateDirection := myMigrater.MigrationDirection(postgresURI, *dbVersion, *step)
+	preMigrationStep, _, _ := myMigrater.Version()
+	for migrateDirection != 0 {
+		migraterErr = myMigrater.Steps(int(migrateDirection))
+		if migraterErr != nil {
+			log.Fatal(migraterErr)
+		}
+		totalMove += int(migrateDirection)
+		log.Printf("Migrated %s by %d step", migrateDirection.String(), intAbs(int(migrateDirection)))
+		migrateDirection = myMigrater.MigrationDirection(postgresURI, *dbVersion, *step)
+	}
+	postMigrationStep, dirty, err := myMigrater.Version()
+	if err != nil {
+		log.Fatal("Failed to get Migration Step for confirmation")
+	}
+	if totalMove == 0 {
+		log.Println("No Changes")
+	} else {
+		log.Printf("Moved %d step(s) in total from Step %d to Step %d", intAbs(totalMove), preMigrationStep, postMigrationStep)
+	}
+
+	currentDBVersion, err := getCurrentDBVersion(postgresURI)
+	if err != nil {
+		log.Fatal("Failed to get Current DB version for confirmation")
+	}
+	log.Printf("DB Version: %s, Migration Step # %d, Dirty: %v", currentDBVersion, postMigrationStep, dirty)
+}
+
+func intAbs(x int) int {
+	return int(math.Abs(float64(x)))
+}
+
+func getCurrentDBVersion(crdbURI string) (string, error) {
+	crdb, err := dssCockroach.Dial(crdbURI)
+	defer func() {
+		crdb.Close()
+	}()
+	if err != nil {
+		return "", fmt.Errorf("Failed to dial CRDB while getting DB version: %v", err)
+	}
+	// check if schema_versions table exists
+	const checkTableQuery = `
+		SELECT EXISTS (
+  		SELECT *
+		  FROM information_schema.tables 
+		WHERE table_name = 'schema_versions'
+	)`
+	var (
+		version = "v0.0.0"
+		exists  bool
+	)
+
+	if err := crdb.QueryRow(checkTableQuery).Scan(&exists); err != nil {
+		return "", err
+	}
+
+	if !exists {
+		return version, nil
+	}
+	// query for the schema version string
+	const getVersionQuery = `
+      SELECT schema_version 
+      FROM schema_versions
+	  WHERE onerow_enforcer = TRUE`
+	if err := crdb.QueryRow(getVersionQuery).Scan(&version); err != nil {
+		return "", err
+	}
+	// if for some reason the string returned is empty
+	if version == "" {
+		version = "v0.0.0"
+	}
+	return version, nil
+}
+
+func parseVersion(version string) [3]int {
+	var (
+		major, minor, patch int
+	)
+	if n, err := fmt.Sscanf(version, "v%d.%d.%d", &major, &minor, &patch); err != nil || n != 3 {
+		log.Panic(err)
+	}
+	result := [3]int{major, minor, patch}
+	return result
+}
+
+// MigrationDirection reads our custom DB version string as well as the Migration Steps from the framework
+// and returns a signed integer value of the Direction and count to migrate the db
+func (m *MyMigrate) MigrationDirection(dbURI string, desiredVersion string, desireStep int) Direction {
+	if desireStep != 0 {
+		migrateStep, dirty, err := m.Version()
+		if err != migrate.ErrNilVersion && err != nil {
+			log.Panic(err)
```

Comment:
I think we want to [return the error here](https://github.com/interuss/dss/pull/328#discussion_r442494747) (and below in this function)

Reply:
I'll have to handle the error and log.Panic outside anyway but I guess its good convention